org.osgi.test.common/src/main/java/org/osgi/test/common/context/CloseableBundleContext.java

```diff
@@ -228,13 +228,23 @@ private static boolean delegatedEquals(Object proxy, Object[] args) {
 	}
 
 	public Bundle installBundle(String location, InputStream input) throws BundleException {
-		Bundle bundle = bundleContext.installBundle(location, input);
+		Bundle bundle = bundleContext.getBundle(location);
+		if (bundle != null) {
```

Comment:
Why is this needed? BundleContext.installBundle will return the existing Bundle object if a bundle with the specified location is already installed.

Reply:
The ERROR in the unit-tests of the last commit shows that multiple instances of the same bundle exists.

Reply:
couldn't `bundlesToBeUninstalledOnClose` just be a `Set`? AFAIK bundles meet equality requirements for it.

Reply:
@rotty3000  you are right. it is a set. fixed in in the src